### PR TITLE
fix(Scripts/Shattrath): Daily Dungeon quests holograms

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

We update once Quest gives AI updates and then once (every 60s) on when 1 or more players are within 30y.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15322
- Closes https://github.com/chromiecraft/chromiecraft/issues/5144

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

<details>

image

image

</details>

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

.go c id 24854
Compare daily HC/N quest to the 2 images
Edit pool_quest_save in characters DB
```sql 
SELECT * FROM `pool_quest_save` WHERE `pool_id` IN (356,357);
```
Edit them with questIDs from the script (356H, 357N)
Should update displayID (when more than 1 player is within 30y, 60s cd)

## Known Issues and TODO List:

- [x] How can I make this rerun on new pool_quest_save (daily reset)?

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/zone_shattrath_city.cpp [hunk 1]
```diff
@@ -253,9 +254,171 @@ class npc_kservant : public CreatureScript
...
+                if (CreatureTemplate const* ci = sObjectMgr->GetCreatureTemplate(templateID))
+                {
+                    CreatureModel const* model = ObjectMgr::ChooseDisplayId(ci);
+                    me->SetDisplayId(model->CreatureDisplayID, model->DisplayScale);
+                }
+            }
+        }
+    }
+
+    void UpdateAI(uint32 /*diff*/) override {}
```
Comment (Nyeriah) [inline src/server/scripts/Outland/zone_shattrath_city.cpp:415]: is this a leftover?

File: src/server/scripts/Outland/zone_shattrath_city.cpp [hunk 2]
```diff
@@ -253,9 +254,171 @@ class npc_kservant : public CreatureScript
...
+    npc_shattrath_daily_quest(Creature* c) : NullCreatureAI(c) {}
+
+    void DoAction(int32 action) override
+    {
+        if (action == ACTION_UPDATE_QUEST_STATUS)
+        {
+            if (!me->GetEntry())
+                return;
+
+            uint32 creature = me->GetEntry();
```
Comment (Nyeriah) [inline src/server/scripts/Outland/zone_shattrath_city.cpp:308]: is this variable needed? you are still using me->GetEntry() in the rest of the script

File: src/server/scripts/Outland/zone_shattrath_city.cpp [hunk 3]
```diff
@@ -253,9 +254,171 @@ class npc_kservant : public CreatureScript
...
+
+struct npc_shattrath_daily_quest : public NullCreatureAI
+{
+    npc_shattrath_daily_quest(Creature* c) : NullCreatureAI(c) {}
+
+    void DoAction(int32 action) override
+    {
+        if (action == ACTION_UPDATE_QUEST_STATUS)
+        {
+            if (!me->GetEntry())
```
Comment (Nyeriah) [inline src/server/scripts/Outland/zone_shattrath_city.cpp:305]: this IF will never return false, as creatures will always have an entry

File: src/server/scripts/Outland/zone_shattrath_city.cpp (hunk 1 above)
Comment (avarishd) [reply src/server/scripts/Outland/zone_shattrath_city.cpp:415]: No, do nullcreatures updateAI?

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**